src/main/java/seedu/dukeofbooks/data/book/BorrowableItem.java

```diff
@@ -0,0 +1,121 @@
+package seedu.dukeofbooks.data.book;
+
+import java.time.LocalDate;
+
+import seedu.dukeofbooks.data.exception.DukeOfBooksException;
+import seedu.dukeofbooks.data.exception.DuplicateActionException;
+import seedu.dukeofbooks.data.person.Person;
+
+
+public abstract class BorrowableItem {
+    private static final String AVAILABLE_STATUS_FORMAT =
+            "Status: Not borrowed";
+    private static final String BORROWED_STATUS_FORMAT =
+            "Status: Borrowed (borrower: %s, due: %s)";
+    private static final String OVERDUE_STATUS_FORMAT =
+            "Status: Overdue (borrower: %s, due: %s)";
+
+    private static final int DEFAULT_RENEW_DAYS = 30;
+    protected boolean isBorrowed;
+    protected LocalDate borrowDate;
+    protected LocalDate dueDate;
+    protected Person borrower;
+
+    public BorrowableItem() {
+        isBorrowed = false;
+        borrowDate = null;
+        dueDate = null;
+        borrower = null;
+    }
+
+    public boolean isBorrowed() {
+        return isBorrowed;
+    }
+
+    LocalDate getBorrowDate() {
+        return borrowDate;
+    }
```

Comment:
I have made a loan class as an association class to keep track of this. Books should not be storing these data since it should only be concerned with data relating to itself.

Loan class will associate book and person as a form of borrow record